deployment/data-streams/changeset/jd_register_nodes.go

```diff
@@ -66,14 +66,20 @@ func validateNodeSlice(nodes []NodeCfg, nodeType string, donIndex int) error {
 	return nil
 }
 
-func registerNodesForDON(e deployment.Environment, nodes []NodeCfg, baseLabels []*ptypes.Label, nodeType NodeType) {
+func registerNodesForDON(e deployment.Environment, donName string, donID int, nodes []NodeCfg, baseLabels []*ptypes.Label, nodeType NodeType) {
 	ntStr := nodeType.String()
 	for _, node := range nodes {
 		labels := append([]*ptypes.Label(nil), baseLabels...)
+
 		labels = append(labels, &ptypes.Label{
 			Key:   "nodeType",
 			Value: &ntStr,
 		})
+
+		labels = append(labels, &ptypes.Label{
+			Key: fmt.Sprintf("don-%d-%s", donID, donName),
```

Comment:
This key feels off. Isn't that a value that we should assign to a key?

Reply:
Yeah, alternatively, we could use a `don` key and assign `fmt.Sprintf("don-%d-%s", donID, donName)` as its value. However, this approach would limit us to a single value per key. For instance, if you wanted to label a node with two DONs, there wouldn’t be a way to attach multiple values to the same key.

Reply:
I suppose whether this is a limitation for data-streams depends on how the product is operated, but it might still be a good idea to future-proof the implementation regardless.